src/rules/curlyRule.ts

```diff
@@ -129,12 +140,64 @@ class CurlyWalker extends Lint.AbstractWalker<Options> {
+            !(this.options.ignoreSameLine && sameLine)) {
             const token = node.getChildAt(childIndex, this.sourceFile);
             const tokenText = ts.tokenToString(token.kind)!;
-            this.addFailure(token.end - tokenText.length, end, Rule.FAILURE_STRING_FACTORY(tokenText));
+            this.addFailure(
+                token.end - tokenText.length, end, Rule.FAILURE_STRING_FACTORY(tokenText),
+                this.addBraceReplacement(statement, node, sameLine));
+        }
+    }
+
+    /** Generate the necessary replacement to add braces to a statement that needs them. */
+    private addBraceReplacement(statement: ts.Statement, node: ts.IterationStatement | ts.IfStatement, sameLine: boolean) {
+        if (sameLine) {
+            return [
+                Lint.Replacement.appendText(statement.getStart(), "{ "),
+                Lint.Replacement.appendText(statement.getEnd(), " }"),
+            ];
+        } else {
+            let positionToAddOpenBrace;
+            let positionToAddCloseBrace = statement.getEnd();
+
+            // This is a bit messy. In order to be sure we add the curly brace
+            // in the right place, we have to find the close paren token after
+            // the condition (or just the 'do' keyword token for do statements).
+            // That token's position depends on the type of statement.
+            if (isIfStatement(node) || isWhileStatement(node)) {
+                // Child 3: close paren on if or while statement
+                positionToAddOpenBrace = node.getChildAt(3).getEnd();
+            } else if (isForStatement(node)) {
+                // Child 7: close paren on for statement
+                positionToAddOpenBrace = node.getChildAt(7).getEnd();
+            } else if (isForInStatement(node) || isForOfStatement(node)) {
+                // Child 5: close paren on for/in or for/of statement
+                positionToAddOpenBrace = node.getChildAt(5).getEnd();
+            } else if (isDoStatement(node)) {
+                // First token: 'do' keyword
+                positionToAddOpenBrace = node.getFirstToken().getEnd();
+                // Child 2: 'while' keyword
+                positionToAddCloseBrace = node.getChildAt(2).pos;
+            } else {
+                throw new Error("Unexpected kind of statement");
+            }
+
+            let indentation = "";
+            // match[1] will be the whitespace at the start of the first
+            // line that isn't all whitespace.
+            const match = /\s*\n([\t ]*)\S/.exec(node.getFullText());
+            if (match != undefined) {
+                indentation = match[1];
+            }
+
+            return [
+                Lint.Replacement.appendText(
+                    this.sourceFile.getLineEndOfPosition(positionToAddOpenBrace), " {"),
+                Lint.Replacement.appendText(statement.getEnd(), `\n${indentation}}`),
```

Comment:
we should try to infer the correct line ending. See https://github.com/palantir/tslint/blob/02dd308c12d0fd4dd6c1b87363357e3ea642da53/src/rules/eoflineRule.ts#L49 or https://github.com/palantir/tslint/blob/02dd308c12d0fd4dd6c1b87363357e3ea642da53/src/rules/orderedImportsRule.ts#L365-L377

Reply:
My years away from Windows have softened me.

I tried to add a test but the test system seems to balk at CRLFs.

Reply:
There are already some tests using CRLF. At least the ones in `test/rules/linebreak-style/*`.
The line endings may be overridden by your editor because of `.editorconfig` or by `git` because of `.gitattributes`.
Your code looks good. I don't think this test case is super important.